lesson_02_homework.rb

```diff
@@ -1,46 +1,154 @@
 
-def full_name(f, l)
-  puts f + l
-end
 
-puts full_nam("bubba", "chuck")
+# ```
+
+# ### Fix Broken Code
 
+# The following code contains some errors and some coding ettiquette issues. 
+# Fix and refactor the code so that it works and is more clear.
 
-# Coding
+# ```ruby
 
+# This is the original
 
-## Enter your code for your 'add' method below
+# def full_name(f, l)
+#   puts f + l
+# end
+# puts full_nam("bubba", "chuck")
 
 
+# This is the correction
 
+# define method full_name (with 2 parameters)
+# change (f, l) to (first_name, last_name)
+# not technically an error but should have variables that make sense
+def full_name(first_name, last_name)
+# set name to nil
+    name = nil
+# code in where names are passed
+# puts should go after end
+    name = first_name + " " + last_name
+end
 
-## Enter the code for you 'join_strings' method below
+# typo: missing e on full_name
+# names were in wrong order...should be "chuck", "bubba"
+puts full_name("chuck", "bubba")
 
 
+# ```
 
+# ### Coding
 
+# 1. Write a method called `add` that does the following:
+#   - takes two parameters and adds them together returning the result. 
+#   For example `add(1, 2)` should return `3`. 
+#   - Add a line that calls your add function with two values, stores the 
+#   result in a variable, and then prints the result to the screen.
+
+# define method add that has 2 parameters
+def add(num1, num2)
+# add parameters num1 and num2
+  num1 + num2
+end
+# pass value to parameter 1 (num1 = 5), pass value to parameter 2 (num2 = 7), print the added values of num1 and num2 and print
+puts add(5, 7)
+ 
+# define method add that has 2 parameters
+def add(num1, num2)
+# add parameters num1 and num2 and store the value in variable sum
+  sum = num1 + num2
+# print the value of sum
+  puts sum
+end
+
+
+# 2. Write a method, `join_strings` that does the following: 
+#   - takes two string values as parameters and joins them together with a space in between, 
+    # returning the result. 
+#   - For example `join_strings("hello", "dolly")` should return `"hello dolly"`.
+
+# define method that concatenates 2 strings ("Ziggy" and "Bigcat")
+def join_strings(firstName='Ziggy', lastName='Bigcat')
+# concatenate of variable firstName and variable lastName 
+# with a space band after firstName (before the curly brace)
+  "#{firstName } #{lastName}"
+end
+# print the concatenated strings
+puts join_strings
+
+
+# 3. Write a method called 'old_enough_to_vote?' that does the following:
+#   - Takes a integer year as a parameter and returns a "yes" if a person born in that year 
+#     is old enough to vote or returns "no" if the person is not old enough to vote
+
+# define method using conditionals around a person's age, based on value in variable year
+def oldEnough (year)
+# condition: if year is less than or equal to 1998
+  if year <= 1998
+# person is old enough to vote, 
+# print out 'You are allowed to vote'
+    puts 'You are allowed to vote'
```

Comment:
🔨 @NancyGoodman  using puts in the method body makes the method less flexible - if you remove the puts you will achieve the same result without forcing the method to print information out to the console.